aws/src/main/java/org/apache/iceberg/aws/s3/S3InputStream.java

```diff
@@ -195,14 +230,20 @@ private void openStream() throws IOException {
     }
   }
 
-  private void closeStream() throws IOException {
+  private void closeStream(boolean closeQuietly) throws IOException {
     if (stream != null) {
       // if we aren't at the end of the stream, and the stream is abortable, then
       // call abort() so we don't read the remaining data with the Apache HTTP client
       abortStream();
       try {
         stream.close();
       } catch (IOException e) {
+        if (closeQuietly) {
+          stream = null;
+          LOG.warn("An error occurred while closing the stream", e);
+          return;
+        }
```

Comment:
This is needed so that IOException failures on close during the retry don't count against the retry on the read. It can just be a best effort close.